types/yandex-metrika-tag/index.d.ts

```diff
@@ -0,0 +1,100 @@
+// Type definitions for non-npm package Yandex.Metrika Tag API 2.0
+// Project: https://yandex.ru/support/metrica/code/counter-initialize.html
+// Definitions by: hikiko4ern <https://github.com/hikiko4ern>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare var ym: ym.Event;
+
+declare namespace ym {
+    interface Event {
+        (counterId: number, eventName: 'init', parameters: InitParameters): void;
+        (counterId: number, eventName: 'addFileExtension', extensions: string | string[]): void;
+        // tslint:disable-next-line no-unnecessary-generics
+        <CTX extends any>(counterId: number, eventName: 'extLink', url: string, options?: ExtLinkOptions<CTX>): void;
+        // tslint:disable-next-line no-unnecessary-generics
+        <CTX extends any>(counterId: number, eventName: 'file', url: string, options?: FileOptions<CTX>): void;
+        (counterId: number, eventName: 'getClientID', cb: (clientID: string) => void): void;
+        // tslint:disable-next-line no-unnecessary-generics
+        <CTX extends any>(counterId: number, eventName: 'hit', url: string, options?: HitOptions<CTX>): void;
+        /** @deprecated */
+        (
+            counterId: number,
+            eventName: 'hit',
+            url: string,
+            title?: string,
+            referer?: string,
+            params?: VisitParameters,
+        ): void;
+        // tslint:disable-next-line no-unnecessary-generics
+        <CTX extends any>(counterId: number, eventName: 'notBounce', options?: NotBounceOptions<CTX>): void;
```

Comment:
You shouldn't need a generic here at all -
```suggestion
        (counterId: number, eventName: 'notBounce', options?: NotBounceOptions<any>): void;
```
should do - same with the other unnecessary generics in this file. Minimally, that'll behave in the same way, without triggering the lint error. Also, prefer `CTX extends {}`, `CTX extends object`, or `CTX extends unknown` (or just unconstrained `CTX`) where possible - the behavior of `T extends any` in old TS versions is strange.

Reply:
Without generics, the context `CTX` type is lost, and for this declaration it will always be `any` -

```ts
ym(counterID, 'notBounce', {
    ctx: 123,
    callback() {
        this; // "any", but "number" is expected
    }
});
```

I removed `extends any`.